### PR TITLE
DFTHG # 977 Document Requirements Expiring Based on Calendar Date

### PR DESCRIPTION
**Pull request by shruthi-rapala**
Updated the new calculation for DPS uplaod date

### CODE REVIEW

File: cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs [hunk 1]
```diff
@@ -4137,9 +4137,19 @@ public DateTimeOffset GetDpsProcessingDate()
...
+            // DFTDP # 977 Change made on 05/02/2024
+
             var mostRecentRecord = dynamicsContext.bcgov_documenturls
-                .Where(x => x.dfp_uploadeddate != null)
-                .OrderByDescending(i => i.dfp_uploadeddate)
+                .Where(x => x.dfp_uploadeddate != null 
+                && x.dfp_submittalstatus == 100000010 // Document status Uplaoded
+                && x.dfp_dpsclass.Contains("9 - General") // Check for 9 - General
+                && x.dfp_DocumentTypeID.dfp_businessarea != 100000001) // Does not contain remedial
+                .OrderBy(i => i.dfp_uploadeddate)
```
Comment (GeorgeWalker) [inline cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs:4152]: I expect this should be OrderByDescending - do we want the most recent date or the oldest?

File: cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs [hunk 2]
```diff
@@ -4159,19 +4169,10 @@ public DateTimeOffset GetDpsUploadedDate()
...
-            // DFTDP # 977 Change made on 05/02/2024
-
             var dpsUplaodedDate = GetDpsUploadedDate();
             var query = from bcgov_documenturl
                         in dynamicsContext.bcgov_documenturls
-                        where bcgov_documenturl.dfp_submittalstatus == 100000010 // Uploaded
-                        && bcgov_documenturl.dfp_dpsclass.Contains("9-General") 
-                        && bcgov_documenturl.dfp_DocumentTypeID.dfp_businessarea != 100000001 // Does not include Remedial document Type
+                        where bcgov_documenturl.dfp_submittalstatus == 100000000 // Open Required                                                  
                         && bcgov_documenturl.dfp_compliancedate < dpsUplaodedDate
```
Comment (GeorgeWalker) [inline cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs:4176]: we should fix the typo "dpsUplaodedDate"

File: cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs (hunk 1 above)
Comment (shruthi-rapala) [reply cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs]: Yes the criteria is to order by oldest uploaded date.

File: cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs (hunk 2 above)
Comment (shruthi-rapala) [reply cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs]: Fixed the typo

**Pull request merged**